gdnative-core/src/core_types/color.rs

```diff
@@ -72,3 +240,57 @@ fn color_repr() {
     use std::mem::size_of;
     assert_eq!(size_of::<Color>(), size_of::<sys::godot_color>());
 }
+
+#[test]
+fn color_to_pixel_color_formats() {
+
+    let color = Color::rgba(1.0, 0.5, 1.0, 0.0);
+    assert_eq!(0xFF7FFF00, color.to_rgba32());
+    assert_eq!(0xFFFF7FFFFFFF0000, color.to_rgba64());
+    assert_eq!(0x00FF7FFF, color.to_abgr32());
+    assert_eq!(0x0000FFFF7FFFFFFF, color.to_abgr64());
+    assert_eq!(0x00FF7FFF, color.to_argb32());
+    assert_eq!(0x0000FFFF7FFFFFFF, color.to_argb64());
+}
+
+godot_test!(test_color {
+    // Test to_html
+    assert_eq!("ffffffff", Color::rgba(1.0, 1.0, 1.0, 1.0).to_html(true).to_string());
+    assert_eq!("ffffff", Color::rgba(1.0, 1.0, 1.0, 1.0).to_html(false).to_string());
+    assert_eq!("80ffffff", Color::rgba(1.0, 1.0, 1.0, 0.5).to_html(true).to_string());
+    assert_eq!("ffffff", Color::rgba(1.0, 1.0, 1.0, 0.5).to_html(false).to_string());
+    assert_eq!("ff8000", Color::rgb(1.0, 0.5, 0.0).to_html(false).to_string());
+    assert_eq!("ff0080ff", Color::rgb(0.0, 0.5, 1.0).to_html(true).to_string());
+    // Test Gray
+    // String comparison due to non-trivial way to truncate floats
+    use crate::core_types::IsEqualApprox;
+    assert!(0.4f32.is_equal_approx(Color::rgb(0.2, 0.4, 0.6).gray()));
+    assert!(0.5f32.is_equal_approx(Color::rgb(0.1, 0.5, 0.9).gray()));
+    assert!(0.9f32.is_equal_approx(Color::rgb(1.0, 1.0, 0.7).gray()));
+    assert!(0.42f32.is_equal_approx(Color::rgb(0.6, 0.6, 0.06).gray()));
+    // Test invert
+    let inverted = Color::rgb(1.0, 1.0,1.0).inverted();
+    assert!(0f32.is_equal_approx(inverted.r));
+    assert!(0f32.is_equal_approx(inverted.g));
+    assert!(0f32.is_equal_approx(inverted.b));
+
+    let inverted = Color::rgb(0.95, 0.95,0.95).inverted();
+    assert!(0.05f32.is_equal_approx(inverted.r));
+    assert!(0.05f32.is_equal_approx(inverted.g));
+    assert!(0.05f32.is_equal_approx(inverted.b));
+
+    let inverted = Color::rgb(0.05, 0.95,0.55).inverted();
```

Comment:
Maybe run `cargo fmt`, to fix formatting errors like this one 🙂

Reply:
Oops. Sorry about that. I'll fix that up.